sql/2021/privacy/most_common_featurepolicy_permissionspolicy_directive_values.sql

```diff
@@ -0,0 +1,141 @@
+#standardSQL
+# Most common directives for Feature-Policy or Permissions-Policy
+
+WITH page_ranks AS (
+  SELECT
+    client,
+    page,
+    rank
+  FROM
+    `httparchive.almanac.summary_response_bodies`
+  WHERE
+    date = '2021-07-01' AND firstHtml = TRUE
+),
+
+response_headers AS (
+  SELECT
+    client,
+    page,
+    LOWER(JSON_VALUE(response_header, '$.name')) AS header_name,
+    LOWER(JSON_VALUE(response_header, '$.value')) AS header_value
+  FROM
+    `httparchive.almanac.summary_response_bodies`,
+    UNNEST(JSON_QUERY_ARRAY(response_headers)) response_header
+  WHERE
+    date = '2021-07-01' AND firstHtml = TRUE
+),
+
+meta_tags AS (
+  SELECT
+    client,
+    url AS page,
+    LOWER(JSON_VALUE(meta_node, '$.http-equiv')) AS tag_name,
+    LOWER(JSON_VALUE(meta_node, '$.content')) AS tag_value
+  FROM (
+    SELECT
+      _TABLE_SUFFIX AS client,
+      url,
+      JSON_VALUE(payload, "$._almanac") AS metrics
+    FROM
+      `httparchive.pages.2021_07_01_*`
+    ),
+    UNNEST(JSON_QUERY_ARRAY(metrics, "$.meta-nodes.nodes")) meta_node
+  WHERE
+    JSON_VALUE(meta_node, '$.http-equiv') IS NOT NULL
+),
+
+total_nb_pages AS (
+  SELECT
+    client,
+    rank,
+    COUNT(DISTINCT page) AS total_nb_pages
+  FROM
+    `httparchive.almanac.summary_response_bodies`
+  WHERE
+    date = '2021-07-01' AND firstHtml = TRUE
+  GROUP BY
+    1, 2
+),
```

Comment:
Same comment about using `summary_requests` over `summary_response_bodies`

Also not sure you understand how the ranking works.
The `rank` column will be 1,000 (if it's in the top 1,000 websites), 10,000 (of it's in the next 9,900 sites)...etc.

See this query: https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2021/pwa/sw_adoption_over_time_ranking.sql for an example as to how to query this. Basically we create a `rank_grouping` column and then use that to include all URLs where `rank <= rank_grouping`.

Reply:
I think I get the ranking, in that we'll indeed see (10000-1000) sites with `rank = 10000` when we group like in the current query.
I'm thinking it might be interesting though to know the numbers for each 'slice' of ranks, instead of 'duplicating' highly ranked sites? (so having the data for 1-1000, 1001-10000, 10001-100000, ... instead of for 1-1000, 1-10000, 1-100000, ...) That way we distinctly see what's happening for lower-ranked sites.
I'd be interested to hear what the others think about this though.

Reply:
I thought the same initially but changed my mind. I think it's really confusing for a reader to see the "rank" of 10,000 and to have to realise it's actually only the next segment of 9,000 pages, rather than the top 10,000 sites.

And labelling it like this (which is what I initially did), is honest, but still quite confusing:

![image](https://user-images.githubusercontent.com/10931297/134474500-7d431143-456f-448b-8033-dbada7d63680.png)

It also makes it less comparable to other stats (e..g those based on Alexa top 1M)

Plus we should be consistent in this usage across the Web Almanac and, at the moment, we're going with the alternative method (cause we think it's less confusing). But happy to hear other's thoughts on this!

Reply:
Discussed with the other maintainers and we're gonna have to insist on the ranking by top XXX, rather than each slice. For all the reasons I gave above. Will add to the Analysts Guide when I get a second so clear in future.

So would you mind correcting the query with this in mind?